common/config/persistence.go

```diff
@@ -39,8 +39,20 @@ func (c *Persistence) DefaultStoreType() string {
 
 // Validate validates the persistence config
 func (c *Persistence) Validate() error {
-	stores := []string{c.DefaultStore, c.VisibilityStore}
-	for _, st := range stores {
+	if _, ok := c.DataStores[c.DefaultStore]; !ok {
```

Comment:
nit: line 42 - 44 is already covered in the for loop

Reply:
Removed. Good catch, thanks!